src/trafficgen/encounter.py

```diff
@@ -1,736 +1,734 @@
-
-    Params:
-        * target_ship: Target ship data
-        * lat_lon0: Reference point, latitudinal [rad] and longitudinal [rad]
-
-    Returns
-    -------
-        ship: Updated target ship data
-    """
-    assert target_ship.initial is not None
-
-    lat_0 = lat_lon0[0]
-    lon_0 = lat_lon0[1]
-
-    lat, lon, _ = flat2llh(
-        target_ship.initial.position.north,
-        target_ship.initial.position.east,
-        lat_0,
-        lon_0,
-    )
-    target_ship.initial.position.latitude = round(lat, 8)
-    target_ship.initial.position.longitude = round(lon, 8)
-    return target_ship
-
-
-def update_position_data_own_ship(
-    ship: OwnShip,
-    delta_time: float,
-) -> OwnShip:
-    """
-    Update ship data of the own ship to also include waypoints.
-
-    Params:
-        * ship: Own ship data
-        * delta_time: Delta time from now to the time new position is being calculated [sec]
-
-    Returns
-    -------
-        ship: Updated own ship data
-    """
-    assert ship.initial is not None
-
-    lat_0 = ship.initial.position.latitude
-    lon_0 = ship.initial.position.longitude
-
-    ship.initial.position.east = 0
-    ship.initial.position.north = 0
-
-    ship_position_future = calculate_position_at_certain_time(
-        ship.initial.position,
-        ship.initial.sog,
-        ship.initial.cog,
-        delta_time,
-    )
-
-    lat_future, lon_future, _ = flat2llh(
-        ship_position_future.north,
-        ship_position_future.east,
-        lat_0,
-        lon_0,
-    )
-    ship_position_future.latitude = round(lat_future, 8)
-    ship_position_future.longitude = round(lon_future, 8)
-
-    ship.waypoints = [
-        Waypoint(position=ship.initial.position.model_copy()),
-        Waypoint(position=ship_position_future),
-    ]
-
-    return ship
-
-
-def decide_target_ship(target_ships: List[TargetShip]) -> TargetShip:
-    """
-    Randomly pick a target ship from a list of target ships.
-
-    Params:
-        * target_ships: list of target ships
-
-    Returns
-    -------
-        The target ship, info of type, size etc.
-    """
-    num_target_ships: int = len(target_ships)
-    target_ship_to_use: int = random.randint(1, num_target_ships)
-    target_ship: TargetShip = target_ships[target_ship_to_use - 1]
-    return target_ship.model_copy(deep=True)
+"""
+Functions to generate encounters consisting of one own ship and one to many target ships.
+The generated encounters may be of type head-on, overtaking give-way and stand-on and
+crossing give-way and stand-on.
+"""
+
+import random
+from typing import List, Optional, Tuple, Union
+from uuid import uuid4
+
+import numpy as np
+
+from trafficgen.check_land_crossing import path_crosses_land
+from trafficgen.marine_system_simulator import flat2llh
+from trafficgen.types import (
+    EncounterRelativeSpeed,
+    EncounterSettings,
+    EncounterType,
+)
+
+from maritime_schema.types.caga import Initial, OwnShip, Position, TargetShip, Waypoint

```

Comment:
tip: If you put the list of imported types in brackets and add a trailing comma behind the last type, then saving the file will automatically format the list with one-type-per-line